### ISSUE TITLE
Add ability to work with private repositories

### ISSUE THREAD

**Issue by zamotivator** [govend/govend#35]
I need to vendor my own project repositories (for instance, repository 'api' which shared between frontend and backend).
For these repositories I want to specify clone url like git@bitbucket.org/....., instead of https:// scheme which ask my username + password every time when I do something with dependencies (or on simple checkout)

**Comment by jackspirou** [comment]
@zamotivator agreed there should be a solid plan and pattern to handle this case.  Its one of the reasons I started govend, so I could selfishly include my companies private repos: https://github.com/govend/govend/blob/9a085218c2d461217e35df2df4cebec657db72bc/deps/vcs/vcs.go#L863-L869

In your case though, would a simple `netrc` file give you what your looking for?

**Comment by jackspirou** [comment]
Here is some documentation for `.netrc` that you might find helpful in the short term: 
* https://confluence.atlassian.com/bitbucketserver/permanently-authenticating-with-git-repositories-776639846.html
* https://gist.github.com/technoweenie/1072829

**Comment by jackspirou** [comment]
@zamotivator any thoughts on the comments above?  Has the `.netrc` file solved your issue?

**Comment by JohnMaguire** [comment]
@jackspirou I'd prefer not to use `.netrc` as I do not like to store my credentials on my FS unencrypted, especially to Github.

**Comment by tboerger** [comment]
> @jackspirou I'd prefer not to use .netrc as I do not like to store my credentials on my FS unencrypted, especially to Github.

You can also use a token which prevents leaking your real credentials.

**Comment by jackspirou** [comment]
@JohnMaguire as @tboerger mentioned above, you can use a github personal token, in lieu of your password within the `.netrc` file.

I think this is an issue worth noting, given the `.netrc` option, it shouldn't stop anyone at the moment. 

I would like to know how `go get` behaves around private repositories, and private GHE servers.  Does it start with `ssh` and then default to `https` and then to `http`?

**Comment by JohnMaguire** [comment]
Thanks, I did not realize that. I feel much better about using an access token.

As far as handling private repositories go, there is some interesting discussion here: https://gist.github.com/shurcooL/6927554

I also just now realized (d'oh!) the behavior that I thought I was seeing: govend defaulting to HTTPS regardless, is actually because I had a typo in my package name. I am not yet running into the issue seen here. It does however look like the above Gist might contain a workaround.

**Comment by Kane-Sendgrid** [comment]
There is a way to force git to always use ssh: https://www.quernus.co.uk/2015/10/22/force-git-to-use-ssh/
